staging/src/k8s.io/component-base/logs/api/v1/registry.go

```diff
@@ -83,6 +85,8 @@ func newLogFormatRegistry() *logFormatRegistry {
 
 // register adds a new log format. It's an error to modify an existing one.
 func (lfr *logFormatRegistry) register(name string, format logFormat) error {
```

Comment:
I thought this was called only from init() functions, which should be race-free

Reply:
https://github.com/kubernetes/kubernetes/blob/d89d5ab2680bc74fe4487ad71e514f4e0812d9ce/test/integration/controlplane/kube_apiserver_test.go#L534-L547 runs two API servers in parallel in the same process.

Both calls of `kubeapiservertesting.StartTestServerOrDie` invoke the `logFormatRegistry`.

Reply:
We are in good company :stuck_out_tongue_closed_eyes: 

`google.golang.org/grpc/grpclog.SetLoggerV2` works the same way, and has the same problem when our `RunCustomEtcd` gets called more than once.

Reply:
Avoiding writes in `klog.ClearLogger` wouldn't have been viable long-term because I have another PR pending where `component-base/logs` installs a logger also in the default configuration. Instead I opted for making handling of recurring `ValidateAndApply` calls dependent on the binary (actual command, unit test, integration test involving test servers).